File: api/src/main/java/net/blueshell/api/dto/FileDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.common.enums.FileType;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "File")
public class FileDTO extends BaseDTO {
    private Long id;
    private String name;
    private String mediaType;
    private Long size;
    private FileType type;
}

File: api/src/main/kotlin/net/blueshell/api/dto/FileDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import net.blueshell.api.common.enums.FileType
import net.blueshell.api.dto.base.AuditedAutoIdDTO

@Schema(name = "File")
data class FileDTO(
    var name: String? = null,
    var mediaType: String? = null,
    var size: Long? = null,
    var type: FileType? = null
) : AuditedAutoIdDTO()
